packages/react-widgets/src/widgets/TimeSeriesWidget.js

```diff
@@ -0,0 +1,308 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { getTimeSeries } from '../models';
+import { addFilter, removeFilter, selectSourceById } from '@carto/react-redux';
+import { NoDataAlert, TimeSeriesWidgetUI, WrapperWidgetUI, TIME_SERIES_CHART_TYPES } from '@carto/react-ui';
+import {
+  GroupDateTypes,
+  AggregationTypes,
+  _FilterTypes as FilterTypes,
+  _getApplicableFilters as getApplicableFilters
+} from '@carto/react-core';
+import { capitalize, Menu, MenuItem, SvgIcon, Typography } from '@material-ui/core';
+import { PropTypes } from 'prop-types';
+
+const BUCKET_SIZE_RANGE_MAPPING = {
+  [GroupDateTypes.WEEKS]: 60 * 60 * 24 * 7,
+  [GroupDateTypes.DAYS]: 60 * 60 * 24
+};
+
+function TimeSeriesWidget({
+  // Widget
+  id,
+  title,
+  dataSource,
+  column,
+  operationColumn,
+  operation,
+  stepSizeOptions,
+  onError,
+  wrapperProps,
+  noDataAlertProps,
```

Comment:
This widget is the only one that can set noDataAlertProps, is it ok @borja-munoz

Reply:
NoDataAlert is a new component, I think we should bring this improvement to other widgets.

Reply:
It's a bit strange that this widget can offer this possibility and the others not. @borja-munoz @Clebal I think that you should decide if add it to other widgets or remove it from this widget for now

Reply:
CategoryWidgetUI had it, others didn't.

Reply:
It's a matter of time. If you think you can safely add it to the other widgets and we will still meet the deadline for the time series widget, go ahead. If you are not sure, it is better to remove it from this widget for now.

Reply:
Challenge accepted.

Reply:
What do you mean with this 🔝  @Clebal ? Will you add noDataAlertProps here to other widgets?

Reply:
Done in another PR https://github.com/CartoDB/carto-react/pull/188